src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -1799,6 +1807,44 @@ bool core::check_tx_semantic(const transaction& tx, bool keeped_by_block) const
     return true;
 }
 //-----------------------------------------------------------------------------------------------
+void core::check_service_node_ip_address() {
+    if (!m_service_node || m_has_ip_check_disabled || service_node_list.debug_allow_local_ips) {
+        return;
+    }
+
+    auto service_node_ip = epee::string_tools::get_ip_string_from_int32(m_sn_public_ip);
+    auto service_node_address = "tcp://{}:{}"_format(service_node_ip, m_quorumnet_port);
+    auto connection_error_callback = [service_node_address]() {
+        log::warning(
+                globallogcat,
+                "Unable to ping configured service node address ({})!",
+                service_node_address);
+    };
+
+    m_omq->connect_remote(
+            oxenmq::address{service_node_address, tools::view_guts(m_service_keys.pub_x25519)},
+            [&](auto conn) {
```

Comment:
You'll need to capture it by value here, too (otherwise this success callback can fire after the outer function has already returned).

Reply:
Good catch, I've fixed this by capturing the variables by value on 1830

Reply:
The change to 1830 isn't necessary (because `conn=conn` and just `conn` in a lambda capture do the same thing).

The problem is: 1830 is too late to make the copy because the copy happens when the lambda itself get constructed at line 1830, but at that point, it's making a copy of the reference that the 1826 lambda is holding, but *that* reference that may be invalid, so 1830 is making a copy okay, but it's making a copy from the potentially invalid reference that 1826 is holding.

Changing this line to:
```suggestion
            [this, connection_error_callback](auto conn) {
```
should solve it (basically copy into the on_connected callback, and then copy from there into the lambd

Reply:
got it, callbacks in callbacks can get a little confusing 🥴

Reply:
Especially in nested lambdas :)